src/main/java/org/railway/utils/Action.java

```diff
@@ -1,44 +1,87 @@
 package org.railway.utils;
 
-import org.openqa.selenium.By;
-import org.openqa.selenium.JavascriptExecutor;
-import org.openqa.selenium.WebElement;
+import org.openqa.selenium.*;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
 
 public class Action {
-    public static void click(WebElement element) {
-        element.click();
+    public static void click(By element) {
+        WebElement webElement = Driver.driver.findElement(element);
+        scroll(element);
+        webElement.click();
     }
 
     public static WebElement find(By element) {
         return Driver.driver.findElement(element);
     }
 
-    public static String getText(WebElement element) {
-        return element.getText();
+
+    public static void getWait(int time) {
+        Driver.wait = new WebDriverWait(Driver.driver, Duration.ofSeconds(time));
+    }
+
+    public static boolean isDisplayed(By element) {
+        try {
+            WebElement elements = find(element);
+            Driver.wait.until(ExpectedConditions.visibilityOfElementLocated(element));
+            return elements.isDisplayed();
+        } catch (NoSuchElementException | TimeoutException e) {
+            return false;
+        }
+    }
+
+    public static String getText(By element) {
+        return Driver.driver.findElement(element).getText();
     }
 
-    public static void sendKeys(WebElement element, String text) {
-        element.sendKeys(text);
+    public static void sendKeys(By element, String text) {
+        WebElement webElement = Driver.driver.findElement(element);
+        scroll(element);
+        webElement.clear();
+        webElement.sendKeys(text);
     }
 
-    public static void selectByValue(WebElement element, String value) {
-        Select select = new Select(element);
+    public static void selectByValue(By element, String value) {
+        WebElement webElement = Driver.driver.findElement(element);
+        Select select = new Select(webElement);
         select.selectByValue(value);
     }
 
-    public static void selectByVisibleText(WebElement element, String text) {
-        Select select = new Select(element);
+    public static void selectByVisibleText(By element, String text) {
+        WebDriverWait wait = new WebDriverWait(Driver.driver, Duration.ofSeconds(20)); // Increased timeout
+        WebElement webElement = Driver.driver.findElement(element);
+        wait.until(ExpectedConditions.visibilityOf(webElement));
```

Comment:
you can create another method to wait for element visible, later then you can reuse it